### PR TITLE
display code coverage as PR comments

### LINKED ISSUE #3048: Enable code coverage on PRs

### PR DESCRIPTION
**Pull request by westin-m**
Use GitHub Actions to display a code coverage report in the Actions summary and in a PR comment. This does not show a difference based on the proposed code changes. That is a future work item.
Closes #3048 

The coverage report displayed in the comment is also available in the summary of the testing action:
Show all checks (if all have passed and it's collapsed) -> Run unit tests: details -> Summary

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: .github/workflows/dotnetcore.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 name: "Run unit tests"
 
+permissions: write-all
```
Comment (kellyyangsong) [inline .github/workflows/dotnetcore.yml:3]: Is it possible to reduce the scope of this permission?

**Comment by jennyf19** [comment]
can we remove these? 
image

**Comment by westin-m** [comment]
> can we remove these? 

I removed the collector from projects that already had it but weren't previously using it. This should only get rid of the unwanted projects in the report and not affect anything else.

File: .github/workflows/dotnetcore.yml (hunk 1 above)
Comment (westin-m) [reply .github/workflows/dotnetcore.yml]: done

**Review by kellyyangsong — APPROVED** [review]
This bad boy LGTM. Thanks @westin-m

**Pull request merged**